packages/remix-dev/config/routesConvention.ts

```diff
@@ -188,6 +188,8 @@ export function createRoutePath(partialRouteId: string): string | undefined {
 
   if (rawSegmentBuffer === "index" && result.endsWith("index")) {
     result = result.replace(/\/?index$/, "");
+  } else {
+    result = result.replace(/\/$/, "");
```

Comment:
I don't _think_ this has ripple effects but would love confirmation.  This normalizes the `path` value for index versus pathless layout routes.  Currently, index routes lose the trailing slash and pathless layout routes keep it.  I figure `<Route path="nested">` seems more idiomatic (user's don't usually include the trailing slash) so I opted to go that route.

Making sure they match is currently required for the logic to detect collisions in `createHierarchicalRoutes`.